packages/destination-actions/src/destinations/fullstory/request-params.ts

```diff
@@ -160,3 +159,66 @@ export const createUserRequestParams = (settings: Settings, requestBody: Object)
     }
   }
 }
+
+/**
+ * Returns {@link RequestParams} for the V2 Create Events HTTP API endpoint.
+ *
+ * @param settings Settings configured for the cloud mode destination.
+ * @param requestValues Values to send with the request.
+ */
+export const createEventRequestParams = (
+  settings: Settings,
+  requestValues: {
+    userId: string
+    eventName: string
+    properties: {}
+    timestamp?: string
+    useRecentSession?: boolean
+    sessionUrl?: string
+  }
+): RequestParams => {
+  const { userId, eventName, properties: eventData, timestamp, useRecentSession, sessionUrl } = requestValues
+  const defaultParams = defaultRequestParams(settings, `v2beta/events?${integrationSourceQueryParam}`)
+
+  const requestBody: Record<string, any> = {
+    name: eventName,
+    properties: eventData
+  }
+
+  if (userId) {
+    requestBody.user = {
+      uid: userId
+    }
+  }
+
+  if (timestamp) {
+    requestBody.timestamp = timestamp
+  }
+
+  if (sessionUrl) {
+    requestBody.session = {
+      id: decodeURIComponent(sessionUrl.substring(sessionUrl.lastIndexOf('/') + 1))
```

Comment:
How do we feel about the robustness of this session id parsing logic? Should we be validating the format of the given session URL?

Reply:
It should be the result of `FS.getCurrentSessionURL()`. Which has the sessionId as everything after the last `/`. Did you have something in mind?

Reply:
My 2c: I think we should eventually adjust our V2 Server API to accept session URLs for `session.id` as a convenience so that we don't require clients to parse the session replay URLs to pluck out Ids. For now, we're probably safe to assume that the URL from won't change in a way that breaks in the near future. So this shouldn't block the PR.

I'll share a note in #prog-api about this as well.